File: leetcode/python/1894.find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        n = len(chalk)
        full = sum(chalk)
        k %= full
        
        for i in range(n):
            if k < chalk[i]:
                return i
            k -= chalk[i]
        return -1

File: leetcode/cpp/1894.find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        long long total = 0;
        for (auto n : chalk)
            total += (long)n;
        
        long rem = (long)k % total;
        for (int i=0; i < chalk.size(); i++) {
            if (rem < chalk[i]) {
                return i;
            }
            rem -= (long)chalk[i];
        }
        return -1;
    }
};

